uberfire-commons-editor/uberfire-commons-editor-backend/src/main/java/org/uberfire/ext/editor/commons/backend/version/VersionServiceImpl.java

```diff
@@ -48,47 +46,40 @@
     @Inject
     private SessionInfo sessionInfo;
 
-    @Override
-    public List<VersionRecord> getVersions( final Path path ) {
-        try {
-            final List<VersionRecord> records = ioService.getFileAttributeView( convert( path ),
-                                                                                VersionAttributeView.class ).readAttributes().history().records();
-
-            final List<VersionRecord> result = new ArrayList<VersionRecord>( records.size() );
-
-            for ( final VersionRecord record : records ) {
-                result.add( new PortableVersionRecord( record.id(), record.author(), record.email(), record.comment(),
-                                                       record.date(), record.uri() ) );
-            }
+    @Inject
+    private VersionRecordService versionRecordService;
 
-            return result;
+    @Override
+    public List<VersionRecord> getVersions(final Path path) {
 
-        } catch ( Exception e ) {
-            throw new RuntimeException( e );
+        try {
+            return versionRecordService.load(Paths.convert(path));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
If you want the exception to propagate to the UI it might be better to use org.guvnor.common.services.backend.exceptions.ExceptionUtilities.handleException(e) as this returns a @Portable exception to the client which can then show a generic exception popup (if the client uses org.uberfire.ext.widgets.common.client.callbacks.DefaultErrorCallback).